packages/react-dom/src/__tests__/ReactDOMServerPartialHydration-test.internal.js

```diff
@@ -2216,17 +2217,18 @@ describe('ReactDOMServerPartialHydration', () => {
     }
 
     const onEvent = jest.fn();
-    const TestResponder = React.DEPRECATED_createResponder(
-      'TestEventResponder',
-      {
-        targetEventTypes: ['click'],
-        onEvent,
-      },
-    );
 
     function Button() {
-      const listener = React.DEPRECATED_useResponder(TestResponder, {});
-      return <a DEPRECATED_flareListeners={listener}>Click me</a>;
+      const ref = React.useRef(null);
+      const effect = () => {
+        return setClick(ref.current, onEvent);
+      };
+      if (isServerRendering) {
+        React.useEffect(effect);
```

Comment:
What is this for? You might as well omit it since server doesn't call effects.

Reply:
It’s still valid as this is what we do internally. We can’t use useEffect though as the event listeners won’t be added at the right time.

Reply:
I'm still not following. I'm saying that

```js
if (isServerRendering) {
  React.useEffect(effect);
} else {
  React.useLayoutEffect(effect);
}
```

should be completely equivalent to writing

```js
if (!isServerRendering) {
  React.useLayoutEffect(effect);
}
```

because on the server, effects are ignored.

Reply:
Gotcha!